### DISCUSSION TITLE
Issue when use preset Redis in CI

### DISCUSSION THREAD

**Discussion by hungngphtiki** [orlangure/gnomock#307] [Q&A] [answered]
test panicked: can't start container: can't pull image: can't pull image: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

**Comment by orlangure** [comment]
Hi @hungngphtiki, my question would be the same as the one in the error message: Is the docker daemon running?
Which CI tool do you use? Are you sure it is possible to use docker inside your CI jobs? To confirm, run the following line as a part of your CI job, just before go tests: `docker version` and see if it works.

**Reply by hungngphtiki** [reply]
![image](https://user-images.githubusercontent.com/88999467/135109771-660edb32-24b8-4b15-82a9-829cb6641a1a.png)
My CI tool is Jenkins and I'm sure docker works inside my CI jobs. I try "docker build --no-cache" in local but still fail with the same error. I have 2 preset include: Mysql and Redis. Mysql is work, but Redis is not working. You can see the image.

**Reply by hungngphtiki** [reply]
![image](https://user-images.githubusercontent.com/88999467/135110602-02001769-eb98-4567-9952-df823af51b0b.png)
This is my Dockerfile.
![image](https://user-images.githubusercontent.com/88999467/135111139-4c6f7181-1a0b-4bc1-ad68-a40a20de48d5.png)
This is my Makefile.

**Reply by hungngphtiki** [reply]
![image](https://user-images.githubusercontent.com/88999467/135110780-c974cae6-1060-4c87-a3aa-c104faa0d0ea.png)
And this is my test file.

**Comment by orlangure** [comment, marked as answer]
According to your screenshots, you are trying to run go tests inside a docker container that you build from a Dockerfile.

Gnomock needs a running docker daemon (either running locally, or via `DOCKER_HOST` environment variable), and I suppose there is no docker daemon running inside the image built from the attached Dockerfile.

For the tests to work, you need to make sure that wherever `go test` command is executed, `docker version` command also works. Not locally, not generally in jenkins, but exactly in the same shell the you execute `go test` from.
